tensorboard/backend/event_processing/plugin_event_multiplexer.py

```diff
@@ -432,17 +471,3 @@ def GetAccumulator(self, run):
     """
     with self._accumulators_mutex:
       return self._accumulators[run]
-
-
-def GetLogdirSubdirectories(path):
```

Comment:
Hmm, would it be more general to just replace LIstRecursively() with the iterative glob technique, rather than leaving it using Walk() and then getting rid of both copies of GetLogdirSubdirectories()?  That way we don't have to duplicate the code in both multiplexers.

I think it doesn't really add overhead to return all the filenames and then do the filtering in a single pass later (in GetLogdirSubdirectories() like we do now) as long as we keep ListRecursively() as a generator the way it is now.  We could even make GetLogdirSubdirectories() itself a generator too so that it doesn't block on finishing all the globbing, by just maintaining a dict of subdirs and only yielding a subdir for a given file when it's not yet in the dict.

Happy to send a PR myself for this alternative if that would be easiest, just let me know.

Reply:
+1. See updates.